core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -107,17 +105,13 @@ public Configuration getConfiguration() {
 
     @Override
     public Configuration getConfiguration(TableId tableId) {
-      Preconditions.checkArgument(tableId == TABLE_ID, "Expected " + TABLE_ID + " obtained"
-          + " from IteratorEnvironment.getTableId(), but got: " + tableId);
       return tableConf;
     }
 
   }
 
   private static final byte[] EMPTY_BYTES = new byte[0];
   private static final Range EMPTY_RANGE = new Range();
-  private static final String TABLE_NAME = "rfileScanner";
-  private static final TableId TABLE_ID = TableId.of(TABLE_NAME);
```

Comment:
could have TABLE_ID = null and check that argument for `getConfiguration` and `getTableName` is null/TABLE_ID to avoid accepting random values. Argument should be `env.getTableId()`

Reply:
The RFileScanner is client utility where the user supplies the set of input files, configuration, columns they want to fetch, etc. I don't know that there is utility in validating the table id when this is being used outside of a table context.

Reply:
Yeah, this is a weird case where we need to pretend there is a table id when there isn't. I'm fine also not validating it, my idea was just that we should expect `env.getTableId()` as the way of getting the table id when it is needed instead of accepting any table id

Reply:
Does validating the table id and throwing an exception change the behavior of the code between the 2.1.3 and 2.1.4 releases? Are we going to introduce an error into the clients use of our utility by upgrading?

Reply:
Answering my own question - Looking at the 2.1.3 RFileScanner code, IterEnv doesn't override `getServiceEnv`, `getPluginEnv` or `getTableId` so any usage of those methods would throw an UnsupportedOperationException. We could add the validation, or not, it doesn't matter as the methods aren't called in RFileScanner.